### PR TITLE
Various preparations for detecting out of range DAScale values

### PR DESCRIPTION
**Pull request by t-b**

### CODE REVIEW

**Comment by t-b** [comment]
@MichaelHuth Please review. I'm not 100% sure about the Sutter specific fix.

File: Packages/MIES/MIES_DataConfigurator.ipf [hunk 1]
```diff
@@ -1449,12 +1444,16 @@ static Function DC_FillDAQDataWaveForDAQ(string device, STRUCT DataConfiguration
 		if(config[i][%DAQChannelType] == DAQ_CHANNEL_TYPE_TP)
 			tpAmp = s.DACAmp[i][%TPAMP] * s.gains[i]
```
Comment (MichaelHuth) [inline Packages/MIES/MIES_DataConfigurator.ipf:1449]: low prio: I noticed that this line can be moved out of the DAQChannelType if block.

File: Packages/tests/HardwareAnalysisFunctions/UTF_ReachTargetVoltage.ipf
Comment (MichaelHuth) [inline Packages/tests/HardwareAnalysisFunctions/UTF_ReachTargetVoltage.ipf:1]: From the commit message it is not fully clear to me if the only intention here is to introduce a separate stimset for this test. If yes, then why was it increased to 5 sweeps?

File: Packages/MIES/MIES_DataConfigurator.ipf [hunk 2]
```diff
@@ -1441,12 +1441,16 @@ static Function DC_FillDAQDataWaveForDAQ(string device, STRUCT DataConfiguration
...
+		isUnAssociated = IsNaN(headstage)
+
 		if(config[i][%DAQChannelType] == DAQ_CHANNEL_TYPE_TP)
 			tpAmp = s.DACAmp[i][%TPAMP] * s.gains[i]
 			ASSERT(DimSize(s.testPulse, COLS) <= 1, "Expected a 1D testpulse wave")
+			ASSERT(!isUnAssociated, "DA channel must be associated for DAQ_CHANNEL_TYPE_TP")
 
-			/// @todo why not associated here
-			[minLimit, maxLimit] = HW_GetVoltageRange(s.hardwareType, 0)
+			[minLimit, maxLimit] = HW_GetVoltageRange(s.hardwareType, !isUnAssociated)
```
Comment (MichaelHuth) [inline Packages/MIES/MIES_DataConfigurator.ipf:1453]: ok

**Review by MichaelHuth — CHANGES_REQUESTED** [review]
See comments

File: Packages/MIES/MIES_DAC-Hardware.ipf [hunk 3]
```diff
@@ -863,6 +863,30 @@ Function HW_GetDAFifoPosition(string device, variable dataAcqOrTP)
 	endswitch
 End
 
+/// @brief Return the minimum/maximum voltage ranges for the given hardware type
+///
+/// @param hardwareType One of @ref HardwareDACTypeConstants
+/// @param isAssociated For Sutter hardware the voltage range differs for associated channels or unassociated ones
+Function [variable mimum, variable maximum] HW_GetVoltageRange(variable hardwareType, variable isAssociated)
```
[the remark this answers is not in the document]
Comment (t-b) [reply Packages/MIES/MIES_DAC-Hardware.ipf:870]: Introduce HW_GetVoltageRangeDAC/HW_GetVoltageRangeADC and make HW_GetVoltageRange static.

File: Packages/MIES/MIES_DataConfigurator.ipf [hunk 4]
```diff
@@ -1449,12 +1444,16 @@ static Function DC_FillDAQDataWaveForDAQ(string device, STRUCT DataConfiguration
 		if(config[i][%DAQChannelType] == DAQ_CHANNEL_TYPE_TP)
 			tpAmp = s.DACAmp[i][%TPAMP] * s.gains[i]
 			ASSERT(DimSize(s.testPulse, COLS) <= 1, "Expected a 1D testpulse wave")
+
+			/// @todo why not associated here
+			[minLimit, maxLimit] = HW_GetVoltageRange(s.hardwareType, 0)
```
[the remark this answers is not in the document]
Comment (t-b) [reply Packages/MIES/MIES_DataConfigurator.ipf]: Fixed.

File: Packages/MIES/MIES_DAC-Hardware.ipf (hunk 3 above)
[the remark this answers is not in the document]
Comment (t-b) [reply Packages/MIES/MIES_DAC-Hardware.ipf:870]: Fixed. I now have

```
Function [variable minimum, variable maximum] HW_GetVoltageRange(variable hardwareType, variable channelType, variable isAssociated)
```

which IMHO is okay without the convenience wrappers. This function is only called in a few places.

File: Packages/MIES/MIES_DAC-Hardware.ipf [hunk 5]
```diff
@@ -863,6 +863,30 @@ Function HW_GetDAFifoPosition(string device, variable dataAcqOrTP)
...
+
+	switch(hardwareType)
+		case HARDWARE_NI_DAC:
+			ASSERT(NI_DAC_MIN == NI_ADC_MIN || NI_DAC_MAX == NI_ADC_MAX, "Non-matching DAC/ADC min/max values")
```
[the remark this answers is not in the document]
Comment (t-b) [reply Packages/MIES/MIES_DAC-Hardware.ipf:874]: That code does not exist anymore.

File: Packages/MIES/MIES_DataConfigurator.ipf (hunk 1 above)
Comment (t-b) [reply Packages/MIES/MIES_DataConfigurator.ipf:1449]: Done.

**Comment by t-b** [comment]
Thanks for the review. I've pushed a new version, see the attached 
[range-diff.txt](https://github.com/user-attachments/files/16501147/range-diff.txt).

**Review by MichaelHuth — APPROVED** [review]
ok

**Pull request merged**